plugins/abci_indexing/sqs/sqs_client.go

```diff
@@ -0,0 +1,102 @@
+package pluginsqs
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sqs"
+
+	types "github.com/sedaprotocol/seda-chain/plugins/abci_indexing/types"
+)
+
+var queueURLEnvName = "SQS_QUEUE_URL"
+
+type SqsClient struct {
+	queueURL  string
+	sqsClient *sqs.SQS
+}
+
+func (client *SqsClient) sendMessageBatch(batch []*sqs.SendMessageBatchRequestEntry) ([]*sqs.BatchResultErrorEntry, error) {
+	result, err := client.sqsClient.SendMessageBatch(&sqs.SendMessageBatchInput{
+		Entries:  batch,
+		QueueUrl: &client.queueURL,
+	})
+	if err != nil {
+		return nil, err
```

Comment:
The API doc is a little confusing, but I think if `err` is set here, it means that the whole batch has failed. So I guess the first return value should be the whole batch somehow converted into `[]*sqs.BatchResultErrorEntry`. And I don't know how to do that :)

Reply:
I guess the function can just return the number of failed entries or their IDs instead.

Reply:
Hmm, interesting problem :) I think returning `nil` makes sense since the call site should have access to the `batch` slice and the program can choose how to handle it at that point rather than us having to fabricate some attributes in the `sqs.BatchResultErrorEntry` for all messages. Given that this `err` means no messages were delivered (disregarding ugly network problems).

Reply:
Yeah that makes sense.